05/js/app.js

```diff
@@ -0,0 +1,49 @@
+function Student(studentFirstName, studentLastName) {
+  this.firstName = studentFirstName;
+  this.lastName = studentLastName;
+  this.grades = {
+    maths: [4, 6],
+    english: [3],
+  };
+}
+
+const student1 = new Student("Jerzy", "Lis");
+
+Student.prototype.addGrade = function (subject, grade) {
+  if (subject === "english") {
+    this.grades.english.push(grade);
+    console.log(this.grades.english);
+  } else if (subject === "maths") {
```

Comment:
Problem z takim rozwiązaniem jest taki, że jak się pojawi 100 przedmiotów to ta metoda będzie ogromna.
Trzeba pobrać dane "dynamicznie" i można to zrobić tak: 
- `this.grades` - to obiekt zawierające nazwy przedmiotów oraz oceny jako tablica 
- `this.grades[subject]` - to oceny dla konkretnego przedmiotu
- `typeof this.grades[subject] === 'undefined'` - jeśli to prawda to taki przedmiot nie istnieje w liście ocen, trzeba go dodać.

czyli możemy zrobić tak:
```
if(typeof this.grades[subject] === 'undefined') {
  this.grades[subject] = []; // tworzymy pustą tablicę dla nieistniejącego przedmiotu
}

this.grades[subject].push(grade); // dodajemy ocenę, już nie interesuje nas gdzie bo wiemy, żę jeśli ten przedmiot nie istniał do linika wyżęj utorzyliśmy pustą tablicę
```